kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    /// On error `result` will contain device specific error codes
+    fn hash_done(&'a self, result: Result<(), u32>);
+}
+
+pub trait HMAC<'a> {
+    /// Set the client instance which will receive `hash_done()` callbacks
+    fn set_client(&'a self, client: &'a dyn Client<'a>);
+
+    /// Set the hash key
+    fn set_key(&self, key: *const u8, len: usize) -> ReturnCode;
```

Comment:
hmm, why is key is a raw pointer here instead of slice?

Reply:
It should be converted to a slice.